src/bin/ion/commands/beta/generate/utils.rs

```diff
@@ -29,6 +29,17 @@ pub struct Field {
     pub(crate) isl_type_name: String,
 }
 
+/// Represents an anonymous type that can be a part of another type definition.
+/// This will be used by the template engine to add these intermediate data models for anonymous types
+/// in to the parent type definition's module/namespace.
+#[derive(Serialize)]
+pub struct AnonymousType {
+    pub(crate) target_kind_name: String,
+    pub(crate) fields: Vec<Field>,
+    pub(crate) abstract_data_type: AbstractDataType,
+    pub(crate) anonymous_types: Vec<AnonymousType>,
+}
```

Comment:
(🗺️ PR tour) This is a utility that is used to store anonymous types and use it in the templates to render as child classes/structs for a type definition.